pkg/mcs/scheduling/server/apis/v1/api.go

```diff
@@ -281,11 +300,47 @@ func changeLogLevel(c *gin.Context) {
 // @Success  200  {object}  config.Config
 // @Router   /config [get]
 func getConfig(c *gin.Context) {
-	var config *config.Config
 	svr := c.MustGet(multiservicesapi.ServiceContextKey).(*scheserver.Server)
-	config = svr.GetConfig()
-	config.Schedule.MaxMergeRegionKeys = config.Schedule.GetMaxMergeRegionKeys()
-	c.IndentedJSON(http.StatusOK, config)
+	localCfg := svr.GetConfig()
+	if svr.IsServing() {
+		localCfg.Schedule.MaxMergeRegionKeys = localCfg.Schedule.GetMaxMergeRegionKeys()
+		c.IndentedJSON(http.StatusOK, localCfg)
+		return
+	}
```

Comment:
Maybe we need check primary here. Otherwise, there could be an infinite loop

Reply:
why?

Reply:
If the primary is also `!serving` all the time, the primary will keep sending requests to the primary. It is dangerous